ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryRoutingFunctionalTest.java

```diff
@@ -285,6 +302,35 @@ public void shouldQueryStandbyWhenActiveDeadStandbyAliveQueryIssuedToRouter() {
     assertThat(rows_0.get(1).getRow().get().values(), is(ImmutableList.of(KEY, BASE_TIME, 1)));
   }
 
+  @Test(timeout = 60000)
+  public void shouldFilterLaggyServers() {
+    // Given:
+    ClusterFormation clusterFormation = findClusterFormation(REST_APP_0, REST_APP_1, REST_APP_2);
+    waitForClusterToBeDiscovered(clusterFormation.router.right, 3);
+    sendLagReportingRequest(clusterFormation.router.right,
```

Comment:
IIUC you are reporting artificially high lags and testing those servers are filterd out and the query fails? 👍

Reply:
Yeah, exactly.  One query has a threshold just below and it succeeds and the next is just over and it fails.